File: src/main/java/com/kata/board/repository/PostRepository.java
package com.kata.board.repository;

import com.kata.board.entity.Post;
import org.springframework.data.jpa.repository.JpaRepository;

public interface PostRepository extends JpaRepository<Post, Long> {
} // JpaRepository : 기본적인 CRUD가 가능하도록 제공하는 인터페이스

File: src/main/kotlin/com/kata/board/repository/post/PostRepository.kt
package com.kata.board.repository.post

import com.kata.board.dto.PostDto
import com.kata.board.entity.Post

interface PostRepository {
    fun getPost(id: Long): Post

    fun getAllPost(): List<Post>

    fun savePost(postDto: PostDto): Post

    fun deletePost(id: Long)
}